src/composables/transactionList.ts

```diff
@@ -0,0 +1,272 @@
+import { useStore } from 'vuex';
+import { orderBy, uniqBy } from 'lodash-es';
+import { computed, ref } from 'vue';
+import { Encoded, Tag } from '@aeternity/aepp-sdk';
+import {
+  ITransaction,
+  ITransactionsState, ITransactionStateList,
+} from '../types';
+import { useMiddleware } from './middleware';
+import { AEX9_TRANSFER_EVENT, fetchJson } from '../popup/utils';
+import { useAccounts } from './accounts';
+import JsonBig from '../lib/json-big';
+import { useAeSdk } from './aeSdk';
+
+interface IFetchTransactionsParams {
+  limit: number,
+  recent: boolean,
+  address: Encoded.AccountAddress,
+  multipleAccounts?: boolean
+}
+
+const transactions = ref<ITransactionStateList>({});
+
+function getTransactionStateByAccountAddress(address: Encoded.AccountAddress) {
+  return transactions.value[address];
+}
+
+function getOrInitTransactionStateForAccount(address: Encoded.AccountAddress): ITransactionsState {
+  const accountTransactionState = getTransactionStateByAccountAddress(address);
+
+  if (accountTransactionState) {
+    return accountTransactionState;
+  }
+
+  const newAccountTransactionState: ITransactionsState = {
+    loaded: [],
+    nextPageUrl: '',
+    pending: {},
```

Comment:
Currently we are storing the pending transactions to if the user will close the extension (for example and open it), we will still be able to show him his pending transaction, also because transactions might not be returned by the mdw fast enough. If we will not store in this case the user won't see his transaction in the list neither as pending nor as mined and then it will appear.